stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/factory/StreamOfflinePluginFactory.kt

```diff
@@ -71,15 +71,21 @@ public class StreamOfflinePluginFactory(
         }
     }
 
+    private var customRepositoryFactory: RepositoryFactory? = null
+
+    public fun setCustomRepositoryFactory(repositoryFactory: RepositoryFactory) {
```

Comment:
What about injecting a `RepositoryFactory.Provider` on the constructor?
Our customers could need the currentUser to build their Repositories, and we should provide the userScope as well, then they can cancel their jobs when it is not needed anymore.

Reply:
I didn't notice the `ChatClient.withRepositoryFactoryProvider` method.